scripts/dts/python-devicetree/src/devicetree/edtlib.py

```diff
@@ -1394,149 +1401,147 @@ def _init_props(self, default_prop_types: bool = False,
                 if name not in _DEFAULT_PROP_SPECS:
                     continue
                 prop_spec = _DEFAULT_PROP_SPECS[name]
-                val = self._prop_val(name, prop_spec.type, False, False, None,
-                                     None, err_on_deprecated)
-                self.props[name] = Property(prop_spec, val, self)
+                prop = self._node.props.get(name)
+                val = self._prop_val(prop_spec, prop, err_on_deprecated)
+                self.props[name] = Property(name, prop_spec, val, self)
 
     def _init_prop(self, prop_spec: PropertySpec,
                    err_on_deprecated: bool) -> None:
         # _init_props() helper for initializing a single property.
         # 'prop_spec' is a PropertySpec object from the node's binding.
 
         name = prop_spec.name
-        prop_type = prop_spec.type
-        if not prop_type:
+
+        if not prop_spec.type:
             _err(f"'{name}' in {self.binding_path} lacks 'type'")
 
-        val = self._prop_val(name, prop_type, prop_spec.deprecated,
-                             prop_spec.required, prop_spec.default,
-                             prop_spec.specifier_space, err_on_deprecated)
+        if prop_spec.pattern and prop_spec.default is not None:
+            _err(f"pattern property '{name}' in {self.binding_path} "
+                  "contains default value")
```

Comment:
Why disallow this?

Reply:
because a pattern property is always required to be in place. regular properties with defaults can be obviated from devicetree.

Reply:
I see. It took me a while to realize that it wouldn't even be possible to generate defaults for pattern properties.